### PR TITLE
AB#141773 Sdk Create Sample Record

### PR DESCRIPTION
**Pull request by jlKampos**
[AB#141773](https://dev.azure.com/niposoftware/15ce0e91-931d-4fbf-9169-8c3dde412b54/_workitems/edit/141773)

NOTE: not sure how to increment the version number

### CODE REVIEW

File: Library/Services/INfieldSurveySampleService.cs [hunk 1]
```diff
@@ -108,5 +109,13 @@ public interface INfieldSurveySampleService
...
         /// <returns>True if success, false otherwise</returns>
         Task<bool> UpdateAsync(string surveyId, int sampleRecordId, IEnumerable<SampleColumnUpdate> columnsToUpdate);
+
+        /// <summary>
+        /// Creates custom columns for the specified record in the survey
+        /// </summary>
+        /// <param name="surveyId">The id of the survey</param>
+        /// <param name="sampleColumns">The columns to create with the respective values</param>
+        /// <returns></returns>
+        Task<IEnumerable<SampleColumnCreate>> CreateSampleRecord(string surveyId, IEnumerable<SampleColumnCreate> sampleColumns);
```
Comment (AlbertoBelver) [inline Library/Services/INfieldSurveySampleService.cs:119]:
```suggestion
        Task<IEnumerable<SampleColumnCreate>> CreateAsync(string surveyId, IEnumerable<SampleColumnCreate> sampleColumns);
```

File: Library/Services/INfieldSurveySampleService.cs [hunk 2]
```diff
@@ -108,5 +109,13 @@ public interface INfieldSurveySampleService
         /// <param name="columnsToUpdate">The name and values of the columns to be updated</param>
         /// <returns>True if success, false otherwise</returns>
         Task<bool> UpdateAsync(string surveyId, int sampleRecordId, IEnumerable<SampleColumnUpdate> columnsToUpdate);
+
+        /// <summary>
+        /// Creates custom columns for the specified record in the survey
```
Comment (AlbertoBelver) [inline Library/Services/INfieldSurveySampleService.cs:114]: not only custom columns
```suggestion
        /// Creates a single sample record with the specified columns
```

File: Library/Services/INfieldSurveySampleService.cs [hunk 3]
```diff
@@ -108,5 +109,13 @@ public interface INfieldSurveySampleService
...
+        /// </summary>
+        /// <param name="surveyId">The id of the survey</param>
+        /// <param name="sampleColumns">The columns to create with the respective values</param>
+        /// <returns></returns>
```
Comment (AlbertoBelver) [inline Library/Services/INfieldSurveySampleService.cs:118]: Add the returns description

File: Library/Services/Implementation/NfieldSurveySampleService.cs [hunk 4]
```diff
@@ -188,6 +189,20 @@ private Task<int> ClearAsync(string surveyId, List<SampleFilter> filters, IEnume
                 .FlattenExceptions();
         }
 
+        public async Task<IEnumerable<SampleColumnCreate>> CreateSampleRecord(string surveyId, IEnumerable<SampleColumnCreate> sampleColumns)
```
Comment (AlbertoBelver) [inline Library/Services/Implementation/NfieldSurveySampleService.cs:192]:
```suggestion
        public async Task<IEnumerable<SampleColumnCreate>> CreateAsync(string surveyId, IEnumerable<SampleColumnCreate> sampleColumns)
```

**Review by AlbertoBelver — CHANGES_REQUESTED** [review]

File: Library/Services/INfieldSurveySampleService.cs [hunk 5]
```diff
@@ -108,5 +109,13 @@ public interface INfieldSurveySampleService
         /// <param name="columnsToUpdate">The name and values of the columns to be updated</param>
         /// <returns>True if success, false otherwise</returns>
         Task<bool> UpdateAsync(string surveyId, int sampleRecordId, IEnumerable<SampleColumnUpdate> columnsToUpdate);
+
+        /// <summary>
+        /// Creates a single sample record with the specified columns
+        /// </summary>
+        /// <param name="surveyId">The id of the survey</param>
+        /// <param name="sampleColumns">The columns to create with the respective values</param>
+        /// <returns>Returns the list of created columns and their values</returns>
```
Comment (AlbertoBelver) [inline Library/Services/INfieldSurveySampleService.cs:118]:
```suggestion
        /// <returns>Returns the full columns set and their values for the created sample record</returns>
```

File: Library/Services/Implementation/NfieldSurveySampleService.cs (hunk 4 above)
Comment (jlKampos) [reply Library/Services/Implementation/NfieldSurveySampleService.cs]: done

**Review by AlbertoBelver — APPROVED** [review]

**Pull request merged**